Content.Shared/Chemistry/Reaction/ChemicalReactionSystem.cs

```diff
@@ -165,6 +165,13 @@ private List<string> PerformReaction(Entity<SolutionComponent> soln, ReactionPro
             var solution = comp.Solution;
 
             var energy = reaction.ConserveEnergy ? solution.GetThermalEnergy(_prototypeManager) : 0;
+            // DeltaV: allow reactions to preserve data
+            var datas = reaction.Reactants.SelectMany(reactant =>
+                solution.Contents
+                    .Where(reagent => reagent.Reagent.Prototype == reactant.Key)
+                    .Where(reagent => reagent.Reagent.Data is not null)
+                    .SelectMany(reagent => reagent.Reagent.Data!)).ToList();
```

Comment:
1. insane linq usage
2. this is computed even if its just getting ignored when ConserveData is false